google_guest_agent/non_windows_accounts.go

```diff
@@ -320,6 +321,40 @@ func readGoogleUsersFile() (map[string]string, error) {
 	return res, nil
 }
 
+// Enables AuthorizedKeysCommand to be used instead of an AuthorizedKeysFile
+func writeSSHDConfigFile() error {
+	sshConfig, err := ioutil.ReadFile("/etc/ssh/sshd_config")
+	if err != nil {
+		return err
+	}
+	proposed := filterAuthorizedKeyLines(string(sshConfig))
+	if proposed == string(sshConfig) {
+		return nil
+	}
+	file, err := os.OpenFile("/etc/ssh/sshd_config", os.O_WRONLY|os.O_TRUNC, 0777)
+	if err != nil {
+		return err
+	}
+	defer closeFile(file)
+	file.WriteString(proposed)
+	return nil
+}
+
+// Helper function to set AuthorizedKeysCommand, should only run once in entire program execution
+func filterAuthorizedKeyLines(contents string) string {
```

Comment:
Don't you want to search for key tokens only (i.e AuthorizedKeysCommand and AuthorizedKeysCommandUser) and then return the actual matching lines?

Reply:
this function finds the AuthorizedKeysCommand and AuthorizedKeysCommandUser lines, and replaces them to use the google_authorized_keys_guest binary

All other lines should remain unchanged.